troubleshooting/incrementally_configuring_images.md

```diff
@@ -0,0 +1,297 @@
+# Incrementally configuring images
+
+When IUF or SAT `bootprep` are failing to configure an image, this procedure can be used to incrementally configure an image.
+This avoids re-running the build and configuration steps that have succeeded, which would occur if IUF or SAT `bootprep` were re-run.
+
+The following example assumes the user is creating a `Compute` image, but can be adapted for other image types.
+
+## Setting up
+
+1. (`ncn-mw#`) Find the path to the IUF logs printed by the `iuf` command. It will look like this:
+
+   ```text
+   2023-11-01T14:06:36.128255Z INFO All logs will be stored in /etc/cray/upgrade/csm/iuf/cos-products-20231031/log/20231101140636
+   ```
+
+   See `install.log` in this directory when subsequent steps reference the IUF logs.
+
+1. (`ncn-mw#`) Find the SAT logs by looking for a line like the following in the IUF logs:
+
+   ```text
+   2023-11-01T14:58:22.874482Z INFO [prepare-managed-images] END sat-bootprep-run [Failed]
+   2023-11-01T14:58:22.874567Z DBG  [prepare-managed-images] LOG FILE FOR sat-bootprep-run: argo/cos-products-20231031-gdm63-prepare-images-2rkds/cos-products-20231031-gdm63-prepare-images-2rkds-sat-wrapper-2664865581/main.log
+   ```
+
+   The string starting with `argo/` is the key within the `config-data` bucket of the S3 object
+   which contains the log output from `sat bootprep`. Save this S3 key to an environment variable:
+
+   ```bash
+   export LOGS_S3_KEY="argo/cos-products-20231031-gdm63-prepare-images-2rkds/cos-products-20231031-gdm63-prepare-images-2rkds-sat-wrapper-2664865581/main.log"
+   ```
+
+1. (`ncn-mw#`) Download the workflow SAT logs.
+
+   ```bash
+   cray artifacts get config-data "$LOGS_S3_KEY" main.log
+   ```
+
+1. (`ncn-mw#`) Set `BASE_IMAGE_ID`.
+
+   View the log and look for a line the includes `Creation of image`
+
+   ```bash
+   INFO: Creation of image ssi-cne-1.0.0-32-csm-1.5.x86_64-csm-1.5.0.beta37-10 succeeded: ID f5374c4c-8c5a-4b79-a5e5-aef778ed36cd
+   ```
+
+   Set `BASE_IMAGE_ID` to the value of the image id.  It is important that this value is set to the image id and not the image name.
+
+   ```bash
+   export BASE_IMAGE_ID="f5374c4c-8c5a-4b79-a5e5-aef778ed36cd"
+   ```
```

Comment:
````suggestion
   1. View the log and look for a line the includes `Creation of image`

      ```text
      INFO: Creation of image ssi-cne-1.0.0-32-csm-1.5.x86_64-csm-1.5.0.beta37-10 succeeded: ID f5374c4c-8c5a-4b79-a5e5-aef778ed36cd
      ```

   1. Set `BASE_IMAGE_ID` to the value of the image ID.

      It is important that this value is set to the image ID and not the image name.

      ```bash
      BASE_IMAGE_ID="f5374c4c-8c5a-4b79-a5e5-aef778ed36cd"
      ```
````